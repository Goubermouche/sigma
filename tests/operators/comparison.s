i32 main() {
	i32 a = 100;
	i32 b = 100;

	printf("%d < %d  = %d\n", a, b, a < b);
	printf("%d > %d  = %d\n", a, b, a > b);
	printf("%d <= %d = %d\n", a, b, a <= b);
	printf("%d >= %d = %d\n", a, b, a >= b);
	printf("%d == %d = %d\n", a, b, a == b);
	printf("%d != %d = %d\n", a, b, a != b);

  	printf("lit: %d\n", 100 > 200);

	i32* x = cast<i32*>(malloc(10));
	i32* y = cast<i32*>(malloc(10));

	printf("ptr: %d\n", x > y);

	ret 0;
}
