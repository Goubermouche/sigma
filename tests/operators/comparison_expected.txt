100 < 100  = 0
100 > 100  = 0
100 <= 100 = 1
100 >= 100 = 1
100 == 100 = 1
100 != 100 = 0
lit: 0
